### PR TITLE
[GEOS-8624] WFS3: ?resultType=hits support

### PR DESCRIPTION
**Pull request by fernandor777**
This PR adds support for ?resultType=hits parameter in WFS3 module.

https://osgeo-org.atlassian.net/browse/GEOS-8624

### CODE REVIEW

File: src/community/wfs3/src/main/java/org/geoserver/wfs3/kvp/ResultTypeKvpParser.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import net.opengis.wfs20.ResultTypeType;
+import org.geoserver.ows.KvpParser;
+import org.geotools.util.Version;
+
+/**
+ * ResultType parser for WFS 3.0.
+ *
+ * @author Fernando Mino - Geosolutions
+ */
+public class ResultTypeKvpParser extends KvpParser {
```
Comment (aaime) [inline src/community/wfs3/src/main/java/org/geoserver/wfs3/kvp/ResultTypeKvpParser.java]: I believe the gs-wfs ResultTypeKvpParser should already be covering this, is there a need to roll a custom WFS 3 one?

File: src/wfs/src/main/java/org/geoserver/wfs/response/v2_0/HitsOutputFormat.java [hunk 2]
```diff
@@ -34,8 +36,36 @@ public boolean canHandle(Operation operation) {
         return request != null && request.getResultType() == ResultTypeType.HITS;
     }
 
+    public String getMimeType(Object value, Operation operation) throws ServiceException {
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/response/v2_0/HitsOutputFormat.java]: Output formats are normally format specific (with a few exceptions, like XSLT or OGR based format generators, where the format is decided at runtime). Can you roll a superclass and have two small subclasses handling the details of GML vs JSON encoding instead?

**Comment by aaime** [comment]
Legit compile failure due to API change:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project gs-nsg-wfs-profile: Compilation failure
[ERROR] /home/travis/build/geoserver/geoserver/src/community/nsg-profiles/nsg-wfs-profile/src/main/java/org/geoserver/nsg/pagination/random/IndexOutputFormat.java:[83,5] method does not override or implement a method from a supertype
```

**Comment by aaime** [comment]
@fernandor777 what should we do about this one?

**Comment by fernandor777** [comment]
Since this PR is throwing compile failure due to API change, I'll close it until check deeply how to fix it correctly.

**Pull request closed without merging**